geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py

```diff
@@ -0,0 +1,120 @@
+            os.path.join(path.replace(".ui.json", "_sweep.ui.json"))
+        )
+        ifile_sweep.data["line_id_start"] = lines.min()
+        ifile_sweep.data["line_id_end"] = lines.max()
+        ifile_sweep.data["line_id_n"] = len(np.unique(lines))
+        params = SweepParams(ifile_sweep)
+        driver = SweepDriver(params)
+        driver.run()
+
+        with self.workspace.open(mode="r+"):
+            self.collect_results()
+
+        if self.params.cleanup:
+            self.cleanup()
+
+    def cleanup(self):
+        """Remove files associated with the parameter sweep."""
+        path = os.path.join(os.path.dirname(self.workspace.h5file))
+        with open(os.path.join(path, "lookup.json"), encoding="utf8") as f:
+            files = list(json.load(f))
+            for file in files:
+                os.remove(f"{file}.ui.json")
+                os.remove(f"{file}.ui.geoh5")
+
+        os.remove(os.path.join(path, "lookup.json"))
+        os.remove(os.path.join(path, "SimPEG.log"))
+        os.remove(os.path.join(path, "SimPEG.out"))
+        os.remove(
+            os.path.join(
+                path, self.workspace.h5file.replace(".ui.geoh5", "_sweep.ui.json")
+            )
+        )
+
+    @staticmethod
+    def line_files(path):
+        with open(os.path.join(path, "lookup.json"), encoding="utf8") as file:
+            line_files = {v["line_id"]: k for k, v in json.load(file).items()}
+        return line_files
+
+    def collect_results(self):
+        path = os.path.join(os.path.dirname(self.workspace.h5file))
+        files = DirectCurrentPseudo3DDriver.line_files(path)
+        lines = np.unique(self.params.line_object.values)
+        results_group = SimPEGGroup.create(self.workspace, name="Pseudo3DInversion")
+        models_group = ContainerGroup.create(self.workspace, name="Models")
+        data_result = self.params.data_object.copy(parent=results_group)
+
+        data = {}
+        for line in lines:
+            ws = Workspace(f"{files[line]}.ui.geoh5")
+            survey = ws.get_entity("Data")[0]
+            data = self.collect_line_data(survey, data)
+            mesh = ws.get_entity("Models")[0]
+            mesh = mesh.copy(parent=models_group)
+            mesh.name = f"Line {line}"
+
+        data_result.add_data(data)
+        models_group.parent = results_group
+
+    def collect_line_data(self, survey, data):
+
+        for child in survey.children:  # initialize data values dictionary
+            if "Iteration" in child.name and child.name not in data:
+                data[child.name] = {"values": np.zeros(survey.n_cells)}
+
+        ind = None
+        for child in survey.children:  # fill a chunk of values from one line
+            if "Iteration" in child.name:
+                if ind is None:
+                    ind = ~np.isnan(child.values)
+                data[child.name]["values"][ind] = child.values[ind]
+
+        return data
+
+
+if __name__ == "__main__":
```

Comment:
Would be great if we could still call this inversion with the `inversion.driver` instead, like the others. Feels like those methods should be part of the Sweeper class, which could append its methods to the `InversionDriver`.

Reply:
And looks like GEOPY-620 needs come in first, because `sweep` doesn't exist publicly yet.

Reply:
yes, let's defer to next sprint

Reply:
Removed the pseudo3d driver